src/content/blog/astro-db-deep-dive.mdx

```diff
@@ -0,0 +1,77 @@
+
+When you run `astro dev`, Astro DB will:
+
+- Create an empty database at `.astro/data.db`
+- Read your schema from `db/config.ts`
+- Seed the database from `db/seed.ts`
+- Your database client is now ready
+
+The workflow largely mirrors the Content Collections workflow that Astro users already love. Importantly, the database itself (`data.db`) is *not* *persistent*. It's created new from scratch every time you start up the dev server. This gives you *simple, reproducible* one-off databases.
+
+Astro DB brings together the web framework, the schema, the seed file, and the database itself all into a single cohesive story. We even include an ORM for you: Drizzle. We selected Drizzle because it is a type-safe ORM that lets you get as close to the metal as you want, and is also pluggable which allowed us to add our own behaviors on top.
+
+## Going remote with a hosted database
+
+Astro DB includes a hosted libSQL database that you can connect to during local development and in production. Everything is managed for you through our Astro Studio platform. You can create a new database for your project in ~30 seconds.
+
+To achieve the kind of scale we knew we'd need on our platform, we partnered with Turso who maintain libSQL and operate the largest libSQL hosting platform. Their commitment to a “database per tenant” model was a perfect fit for our need to spin up hundreds of thousands of databases, all on-demand.
+
+We also spent some time last year prototyping Cloudflare's D1 product. We liked the vision of the project, but struggled with the added abstraction layer of workers and worker bindings when all we needed was the database. We were also hesitant to build on a proprietary database technology (D1) especially if it meant bundling that tech into Astro itself. Ultimately, we found that D1 wasn't a good fit for our use-case.
+
+## Zero-downtime schema migrations
+
+Your database schema is defined in the project `db/config.ts` file. When you make changes to your schema, you need to push them to your hosted database in a way that eliminates risk of data loss and downtime. So we built `astro db push`
+
+The `push` command was designed to balance ease-of-use while still encouraging best practices that work on production large-scale projects. In Astro DB there are no migration files to manage. Instead, when you run `push` your schema is automatically compared against your hosted production database and any new changes are applied. If those changes can't be added safely or risk any data loss in any way, the changes will not be applied.
+
+This encourages the  ["expand and contract" migration](https://planetscale.com/blog/safely-making-database-schema-changes) strategy in Astro DB applications. Of course, if you're doing rapid development and don't mind resetting your database as needed, you can run `astro db --force-reset` to push up any schema changes you'd like resetting the database as needed. 
```

Comment:
"resetting the database as needed" is a bit verbose, especially because we're using it two times, I rewrote the sentence a bit to make it easier.
```suggestion
This encourages the ["expand and contract" migration](https://planetscale.com/blog/safely-making-database-schema-changes) strategy in Astro DB applications. Of course, if you're doing rapid development and don't mind resetting your database as needed, you can run `astro db --force-reset` to push up any schema changes you'd like, including a database reset.
```